### PR TITLE
feat: Made sure summaries actually give output quantile objectives as well; improved flag docs and tests for complex type series calculation.

### PR DESCRIPTION
**Pull request by bwplotka**
cc @saswatamcode @jmichalek132

### CODE REVIEW

File: metrics/serve.go [hunk 1]
```diff
@@ -242,10 +245,22 @@ func (c *Collector) recreateMetrics(unsafeGetState readOnlyStateFn) {
 		c.nativeHistograms[id] = histogram
 	}
 
+	// Mimic some quantile objectives.
+	objectives := map[float64]float64{}
+	if c.cfg.SummaryObjectives > 0 {
+		parts := 100 / c.cfg.SummaryObjectives
```
Comment (ridwanmsharif) [inline metrics/serve.go:251]: qq: is it valid for summary objectives to be greater than 100?

**Review by ridwanmsharif — APPROVED** [review]
LGTM

File: metrics/serve.go (hunk 1 above)
Comment (bwplotka) [reply metrics/serve.go:251]: No, they should be from 0.0 to 1.0 - you can see I am dividing by 100.0 later on. Using 100 to have cheap integer mitigation of float precision issues when substracting things later on (:

**Review by saswatamcode — COMMENTED** [review]
Sorry missed this! But LGTM!

**Review by jmichalek132 — COMMENTED** [review]
LGTM

**Pull request merged**